### PR TITLE
semih week2

### PR DESCRIPTION
**Pull request by sigma21**
week2

### CODE REVIEW

File: Week2/queryPackage.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+# with result as
+#     (select name,count(distinct language) from languages group by name),
+# create trigger langcheck after insert on languages for each row IF count(DISTINCT language) > 10
+#     THEN set lc_messages = 'alert!';
+#     END IF;
+
+delimiter $$
+CREATE TRIGGER date_trigger
+    AFTER INSERT
```
Comment (unmeshvrije) [inline Week2/queryPackage.js]: This trigger should be fired BEFORE insert.

File: Week2/queryPackage.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        ON languages
+            FOR EACH ROW
+            BEGIN
+                DECLARE message VARCHAR(100);
+#                 DECLARE project_start_date datetime ;
+#                 DECLARE project_end_date datetime ;
+                DECLARE language_count INTEGER ;
+#                 SET project_start_date = (select start_date from projects where proj_no=new.proj_no);
```
Comment (unmeshvrije) [inline Week2/queryPackage.js]: Remove the unnecessary comments. They are hard to parse while grading.

File: Week2/queryPackage.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+
+# with result as
+#     (select name,count(distinct language) from languages group by name),
+# create trigger langcheck after insert on languages for each row IF count(DISTINCT language) > 10
+#     THEN set lc_messages = 'alert!';
+#     END IF;
+
+delimiter $$
+CREATE TRIGGER date_trigger
```
Comment (unmeshvrije) [inline Week2/queryPackage.js]: date_trigger is not a good name for this trigger.

File: Week2/queryPackage.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+                 SET language_count = (select count(distinct language) from languages group by name);
+                IF language_count >= 10
+                THEN
+                    set message = 'Number of languages are now 10 or more.';
+                    SET lc_messages=message; SIGNAL SQLSTATE '45000';
+                END IF;
+            END $$
+delimiter ;
+
+drop trigger date_trigger;
```
Comment (unmeshvrije) [inline Week2/queryPackage.js]: You should not drop the trigger, because it will not be effective otherwise.

File: Week2/week2.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        for (r of results) {
+            console.log(`${r}`)
+        }                                  
+    } catch(error) {
+        console.error(error);
+    }
+    db.end();
+}
+
+
```
Comment (unmeshvrije) [inline Week2/week2.js]: I think you intended to write a menu driven program (see herehttps://www.geeksforgeeks.org/menu-driven-program-using-switch-case-c/)  which runs a while loop and keeps asking users for input.
your program will terminate after giving you the answer of exactly one query.

File: Week2/queryPackage.js [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+
+const languagesInARegion = `with result as
+                    (select country.Name, country.Region, country.Code, countrylanguage.Language
+                        from new_world.country join new_world.countrylanguage
+                        on country.Code = countrylanguage.CountryCode)
+                    select Language from result where Region = ?`
+
+
+const officialLanguagesInRegion = `with result as
```
Comment (unmeshvrije) [inline Week2/queryPackage.js]: The `with result as` syntax is recommended when you want to reuse it in the subqueries. 
In your case, the `result` subquery is only used once in a single query. So it is not necessary to use that alias. The query could be written as 
```
                    select Language from (select country.Name, country.Region, country.Code, countrylanguage.Language
                        from new_world.country join new_world.countrylanguage
                        on country.Code = countrylanguage.CountryCode) where Region = ?`
```

**Comment by wouterkleijn** [comment]
Thanks @unmeshvrije Can you make the necessary changes @sigma21 ?
